src/full/Agda/TypeChecking/Rules/Term.hs

```diff
@@ -1065,6 +1081,80 @@ unquoteTactic tac hole goal k = do
     Left err -> typeError $ UnquoteFailed err
     Right _ -> k
 
+---------------------------------------------------------------------------
+-- * Generalized identifiers
+---------------------------------------------------------------------------
+
+type NameOrMeta = Either MetaId QName
+
+checkGeneralized :: Set.Set QName -> TCM Type -> TCM (Type, Type)
```

Comment:
Please haddockument!
At some point, please include a longer comment giving a survey of the Generalize feature and how the components interact.